CMake/Findmlpack.cmake

```diff
@@ -0,0 +1,220 @@
+# - Find mlpack
+# Find the mlpack C++ library
+#
+# This module will look for mlpack dependencies and then look for mlpack itself
+# if it is here or not.
+# 
+# We are basically inlining the previous ensmallen and Cereal into this file.
+#
```

Comment:
Do you think you can add some documentation here on what variables get set by this file?  (e.g. `MLPACK_INCLUDE_DIRS`, `MLPACK_LIBRARIES`, etc.)  You can take a look at a lot of the other `Find*.cmake` files that get distributed with mlpack for an example---all the ones I've seen document the variables that they set and what they mean, which is super useful.